services/package-json-helpers.js

```diff
@@ -9,13 +20,33 @@ const isDependencyMap = Joi.object()
   )
   .default({})
 
+/**
+ * Joi schema for validating package json object.
+ * Checks if the object has all the dependency types and the dependency types are valid.
+ *
+ * @type {object}
+ */
 const isPackageJsonWithDependencies = Joi.object({
   dependencies: isDependencyMap,
   devDependencies: isDependencyMap,
   peerDependencies: isDependencyMap,
   optionalDependencies: isDependencyMap,
 }).required()
 
+/**
+ * Determines the dependency version based on the dependency type.
+ *
+ * @param {object} attrs - Refer to individual attributes
+ * @param {string} attrs.kind - Wanted dependency type, defaults to prod
+ * @param {string} attrs.wantedDependency - Name of the wanted dependency
+ * @param {object} attrs.dependencies - Map of dependencies
+ * @param {object} attrs.devDependencies - Map of dev dependencies
+ * @param {object} attrs.peerDependencies - Map of peer dependencies
+ * @param {object} attrs.optionalDependencies - Map of optional dependencies
+ * @throws {string} - Error message if unknown dependency type provided
+ * @throws {InvalidParameter} - Error if wanted dependency is not present
+ * @returns {object} Object containing wanted dependency version
```

Comment:
This function does currently return an object, but it doesn't really need to. I expect this is an artefact of some past refactoring. A nice cleanup we could do here would be to update this so it just returns a string and update the handful of places it is consumed accordingly. Also, the return value (`range`) might be an exact version, but it might also be a semver range i.e: this value could be something like "~2.1.6" or ">=3.0.0,<4.0.0" rather than a precise version number so lets explain that in the docstring.

Reply:
Sure, will update this description. I'll raise a new PR for the cleanup to keep it simple.